saleor/app/installation_utils.py

```diff
@@ -12,6 +13,21 @@
 REQUEST_TIMEOUT = 25
 
 
+class AppInstallationError(HTTPError):
+    pass
+
+
+def validate_app_install_response(response: Response):
+    try:
+        response.raise_for_status()
+    except HTTPError as err:
+        try:
+            error_msg = str(response.json()["error"]["message"])
```

Comment:
Shouldn't you use `.get` here? This can fail if these field will not be returned.

Reply:
you expect that the error response from the app will be in defined shape: json: `{"error": {"message": "err_msg"}}`. This is something that should be added to the docs as the app owner need to know the proper response shape.

Reply:
> Shouldn't you use .get here? This can fail if these field will not be returned.

There is the intention for this line to fail if the error response has improper structure or does not contain an error message at all. In such a case, it will fall back to default HTTPError and then default error message. Any idea for a cleaner way to do that?

> you expect that the error response from the app will be in defined shape: json: {"error": {"message": "err_msg"}}. This is something that should be added to the docs as the app owner need to know the proper response shape.

Yes, I will add it to the documentation when it is accepted. Extending error information with context like error code, source, response status_code, etc., is also being discussed. I will probably document it in bulk.

Reply:
> There is the intention for this line to fail if the error response has improper structure or does not contain an error message at all. In such a case, it will fall back to default HTTPError and then default error message. 

Right, I missed `try/except`, shouldn't do review before first coffee 😅 

> Any idea for a cleaner way to do that?

When building an app with a local Saleor instance, I would be happy to get here warning info like "response structure is incorrect: missing error field", something that would give me a more context of what is wrong or what is missing.